src/basic/conf-files.c

```diff
@@ -32,11 +32,12 @@
 #include "macro.h"
 #include "missing.h"
 #include "path-util.h"
+#include "stat-util.h"
 #include "string-util.h"
 #include "strv.h"
 #include "util.h"
 
-static int files_add(Hashmap *h, const char *root, const char *path, const char *suffix) {
+static int files_add(Hashmap *h, const char *suffix, const char *root, unsigned flags, const char *path) {
```

Comment:
I'm not sure about this argument reordrering. Previously, all the `char *` arguments were in the order in which they appear in a filename. Right now it's the last part, then the first, then some flags, and then finally the middle...

Reply:
Ah, OK, it's because you want the output argument to be last. Makes sense.

Reply:
not just that, but ultimately files_add() is just the inner loop function for the other three functions in this file, and they mostly take the same arguments, but used different ordering for no reason.